source/documentation/modules/map.html.md.erb

```diff
@@ -28,78 +67,297 @@ title: sass:map
+    @debug map.has-key($font-weights, "bolder") // false
+  <% end %>
+
+  If `$keys` is not empty, the map targeted for searching is nested within
+  `$map`. From left to right, including `$key` and excluding the last key in
+  `$keys`, these keys form a path of nested maps that leads to the targeted map.
+
+  Returns true if the targeted map contains a value associated with the last key
+  in `$keys`.
+
+  Returns false if it does not, or if any key in `$keys` references a value
+  that is not a map.
+
+  <% example(autogen_css: false) do %>
+    $fonts: (
+      "Helvetica": (
+        "weights": (
+          "regular": 400,
+          "medium": 500,
+          "bold": 700
+        )
+      )
+    );
+
+    @debug map.has-key($fonts, "Helvetica", "weights", "regular"); // true
+    @debug map.has-key($fonts, "Helvetica", "colors"); // false
+    ===
+    $fonts: ("Helvetica": ("weights": ("regular": 400, "medium": 500, "bold": 700)))
+
+    @debug map.has-key($fonts, "Helvetica", "weights", "regular") // true
+    @debug map.has-key($fonts, "Helvetica", "colors") // false
   <% end %>
 <% end %>
 
 
-<% function 'map.keys($map)', 'map-keys($map)', returns: 'list' do %>
-  Returns a comma-separated list of all the keys in `$map`.
+<% function 'map.get($map, $key, $keys...)',
+            'map-get($map, $key, $keys...)' do %>
+  If `$keys` is empty, returns the value in `$map` associated with `$key`.
+
+  If `$map` doesn't have a value associated with `$key`, returns [`null`][].
+
+  <%= partial 'code-snippets/example-map-get' %>
+
+  If `$keys` is not empty, the map targeted for searching is nested within
+  `$map`. From left to right, including `$key` and excluding the last key in
+  `$keys`, these keys form a path of nested maps that leads to the targeted map.
+
+  Returns the value in the targeted map associated with the last key in `$keys`.
+
+  Returns [`null`][] if the map does not have a value associated with the key,
+  or if any key in `$keys` references a value that is not a map.
+
+  <% example(autogen_css: false) do %>
+    $fonts: (
+      "Helvetica": (
+        "weights": (
+          "regular": 400,
+          "medium": 500,
+          "bold": 700
+        )
+      )
+    );
+
+    @debug map.get($fonts, "Helvetica", "weights", "regular"); // 400
+    @debug map.get($fonts, "Helvetica", "colors"); // null
+    ===
+    $fonts: ("Helvetica": ("weights": ("regular": 400, "medium": 500, "bold": 700)))
+
+    @debug map.get($fonts, "Helvetica", "weights", "regular") // 400
+    @debug map.get($fonts, "Helvetica", "colors") // null
+  <% end %>
+
+  [`null`]: ../values/null
+<% end %>
+
+
+<% function 'map.set($map, $key, $value)',
+            'map.set($map, $keys..., $key, $value)',
+            returns: 'map' do %>
+  <% heads_up do %>
+    In practice, the actual arguments to `map.set($map, $keys..., $key, $value)`
+    are passed as `map.set($map, $args). They are described here as `$map,
```

Comment:
```suggestion
    are passed as `map.set($map, $args...). They are described here as `$map,
```

This is relevant because you can't pass `$args` by name.